client/boot/common.js

```diff
@@ -355,6 +356,11 @@ const setupMiddlewares = ( currentUser, reduxStore ) => {
 
 			next();
 		} );
+	} else {
+		// Temporary Experiment testing the new ExPlat client
+		loadExperimentAssignment( 'explat_test_aa_calypso_boot' );
+		// Duplicated on purpose
+		loadExperimentAssignment( 'explat_test_aa_calypso_boot' );
```

Comment:
These calls can potentially return a rejected promise and trigger a console warning about unhandled promise rejections. Would be nice to wrap them with `catch( noop )` somehow.

Reply:
Importing the ExPlat library during the boot also inflates the entrypoint code by 10kB which is quite significant. I assume that these tests are only temporary and will be removed after some time?

Reply:
> These calls can potentially return a rejected promise and trigger a console warning about unhandled promise rejections. Would be nice to wrap them with catch( noop ) somehow.

`loadExperimentAssignment` is built to never throw in production:
https://github.com/Automattic/wp-calypso/blob/e57f7084bd853e364bfffdff7072d620eb5d0e95/packages/explat-client/src/create-explat-client.ts#L99-L166

Reply:
> Importing the ExPlat library during the boot also inflates the entrypoint code by 10kB which is quite significant. I assume that these tests are only temporary and will be removed after some time?

So these tests are temporary, but it is likely for tests to be initialised here in the future. Importantly `/lib/abtest` adds about 6kb to `entry-main`, and will be removed in the near future (around a month pending developer buy-in), meaning only a net 3kb will be added.

Reply:
Hmm, I could also possibly remove the dependency on `@automattic/calypso-analytics`, bringing it almost in line with `/lib/abtest`:

<img width="300" alt="Screen Shot 2021-03-03 at 7 55 51 pm" src="https://user-images.githubusercontent.com/971886/109802293-92feb900-7c5a-11eb-8ea1-a735a0381bc6.png">

Reply:
> `loadExperimentAssignment` is built to never throw in production:

Thanks, I didn't notice that before :+1: So, if the `loadExperimentAssignment` REST request fails with a network error, a production instance will get some fallback assignment, while my development instance will break? The failure in the development mode seems a bit uncontrolled: for example, the `useEffect` callback in `useExperiment` will never set `isLoading` back to `false`.

Maybe even the development instance should get a fallback? It seems to me that the difference between production and development is now bigger that necessary.

Reply:
🤔 Maybe it should get a fallback. However I might push back on this for now and [create an issue](593-gh-Automattic/experimentation-platform) so we can proceed with the important A/A tests. The new client isn't released for general consumption and we can always change this behaviour later.